src/azfswatchdog

```diff
@@ -0,0 +1,160 @@
+#!/bin/bash 
+
+#  
+# How often does the watchdog look for unmounts and/or IP address changes for 
+# Blob endpoints. 
+# 
+
+MONITOR_INTERVAL_SECS=5 
+
+# 
+# This stores the map of local IP and share name and external blob endpoint IP. 
+# 
+
+IPMAP="/tmp/azfs-ipmap.txt"
+
+
+# 
+# Watchdog for monitoring unmounts and more importantly change in blob endpoint 
+# addresses possibly as a result of migration. 
+# 
+# TODO: This must be a separate process started by init/systemd. 
+# 
+
+azfs_watchdog() 
+{ 
+    # Disable signal to make this slightly more resilient. 
+    trap "" INT QUIT TSTP 
+
+    while :; do 
+        sleep $MONITOR_INTERVAL_SECS 
+
+        findmnt=$(findmnt --raw --noheading -o MAJ:MIN,FSTYPE,SOURCE,TARGET,OPTIONS -t nfs) 
```

Comment:
any/all commands used in this file and in the mount helper, should be added as dependency in the package file, so that we are guaranteed that those will exist if our package is installed.

Reply:
Making list of all these dependencies [here](https://microsoft.sharepoint.com/teams/AzureStorage/_layouts/OneNote.aspx?id=%2Fteams%2FAzureStorage%2FShared%20Documents%2FXNfsv3%2FOneNote%2FXNfsv3%20Development&wd=target%28Design.one%7C3751F5D8-2F38-4C12-A46A-62E5EB666E6E%2FDistribution%20of%20Linux%20NFS%20Mount%20Helper%7C4084951E-8752-4491-98BB-BB37FE6601F8%2F%29
onenote:https://microsoft.sharepoint.com/teams/AzureStorage/Shared%20Documents/XNfsv3/OneNote/XNfsv3%20Development/Design.one#Distribution%20of%20Linux%20NFS%20Mount%20Helper&section-id={3751F5D8-2F38-4C12-A46A-62E5EB666E6E}&page-id={4084951E-8752-4491-98BB-BB37FE6601F8}&end). Can start discussing when I start working on those package files.